Lib/idlelib/idle_test/test_editor.py

```diff
@@ -1,14 +1,432 @@
+        eq = self.assertEqual
+        eq(ps('_spam'), (0, 'spam'))
+        eq(ps('spam'), (-1, 'spam'))
+        eq(ps('spam_'), (4, 'spam'))
+
+    @mock.patch.object(editor.macosx, 'isCocoaTk')
+    def test_get_accelerator(self, mock_cocoa):
+        ga = editor.get_accelerator
+        eq = self.assertEqual
+        keydefs = {'<<zoom-height>>': ['<Alt-Key-9>'],
+                   '<<open-module>>': ['<Control-Shift-O>'],
+                   '<<cancel>>': ['<Cancel>'],
+                   '<<indent>>': ['<Alt-bracketleft>'],
+                   '<<copy>>': ['<Control-j>', '<Control-c>']}
+
+        mock_cocoa.return_value = False
+        eq(ga(keydefs, '<<paste>>'), '')  # Not in keydefs.
+        eq(ga(keydefs, '<<copy>>'), 'Ctrl+J')  # Control to Ctrl and first only.
+        eq(ga(keydefs, '<<zoom-height>>'), 'Alt+9')  # Remove Key-.
+        eq(ga(keydefs, '<<indent>>'), 'Alt+[')  # bracketleft to [.
+        eq(ga(keydefs, '<<cancel>>'), 'Ctrl+Break')  # Cancel to Ctrl-Break.
+        eq(ga(keydefs, '<<open-module>>'), 'Ctrl+Shift+O')  # Shift doesn't change.
+
+        # Cocoa test.
+        mock_cocoa.return_value = True
+        eq(ga(keydefs, '<<open-module>>'), '')  # Cocoa skips open-module shortcut.
+
+
+class MenubarTest(unittest.TestCase):
+    """Test functions involved with creating the menubar."""
+
+    @classmethod
+    def setUpClass(cls):
+        # Test the functions called during the __init__ for
+        # EditorWindow that create the menubar and submenus.
+        # The class is mocked in order to prevent the functions
+        # from being called automatically.
+        w = cls.mock_editwin = mock.Mock(editor.EditorWindow)
+        w.menubar = tk.Menu(root, tearoff=False)
+        w.text = tk.Text(root)
+        w.tkinter_vars = {}
+
+    @classmethod
+    def tearDownClass(cls):
+        w = cls.mock_editwin
+        w.text.destroy()
+        w.menubar.destroy()
+        del w.menubar, w.text, w
+
+    @mock.patch.object(editor.macosx, 'isCarbonTk')
+    def test_createmenubar(self, mock_mac):
+        eq = self.assertEqual
+        ed = editor.EditorWindow
+        w = self.mock_editwin
+        # Call real function instead of mock.
+        cmb = partial(editor.EditorWindow.createmenubar, w)
+
+        # Load real editor menus.
+        w.menu_specs = ed.menu_specs
+
+        mock_mac.return_value = False
+        cmb()
+        eq(list(w.menudict.keys()),
+           [name[0] for name in w.menu_specs])
+        for index in range(w.menubar.index('end') + 1):
+            eq(w.menubar.type(index), tk.CASCADE)
+            eq(w.menubar.entrycget(index, 'label'),
+               editor.prepstr(w.menu_specs[index][1])[1])
+        # Recent Files added here and not fill_menus.
+        eq(w.menudict['file'].entrycget(3, 'label'), 'Recent Files')
+        # No items added to helpmenu, so the length has no value.
+        eq(w.base_helpmenu_length, None)
+        w.fill_menus.assert_called_with()
```

Comment:
If see this assertion and the next one as testing the implementation rather the behavior, so I suggest removing them.